### PR TITLE
test: Authorization Grant Flow without Redirect URI

### PR DESCRIPTION
**Pull request by strehle**
Add test for this hidden feature thus not to run into a regression

### CODE REVIEW

File: uaa/src/main/webapp/WEB-INF/spring/oauth-clients.xml [hunk 1]
```diff
@@ -82,7 +82,7 @@
                             <entry key="secret" value="loginsecret"/>
                             <entry key="scope" value="openid,oauth.approvals"/>
                             <entry key="authorized-grant-types" value="client_credentials,authorization_code"/>
-                            <entry key="redirect-uri" value="http://localhost/**,http://localhost:7000/**"/>
```
Comment (peterhaochen47) [inline uaa/src/main/webapp/WEB-INF/spring/oauth-clients.xml:85]: What is the purpose of this change? Is this required for the added test?

Comment (strehle) [reply uaa/src/main/webapp/WEB-INF/spring/oauth-clients.xml:85]: it mainly is the revert of my change https://github.com/cloudfoundry/uaa/pull/2403/files#diff-d02078cfeded6b34f4eb0a68b222758f944db9bbe2fa1227249a9c782718cf09R85

the login client is used in our internal forked scenario and it was broken. Because UAA has this hidden feature of redirect uris. It means if you have only one redirect uri defined you can omit the parameter redirect_uri in authorize request.
If you have an array in the client you cannot use this feature anymore therefore removed the 2nd entry

Comment (strehle) [reply uaa/src/main/webapp/WEB-INF/spring/oauth-clients.xml:85]: and create an explicit test for this hidden feature

File: uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/AuthorizationCodeGrantIntegrationTests.java [hunk 2]
```diff
@@ -192,6 +194,34 @@ public void testZoneInactive() {
         assertEquals(HttpStatus.NOT_FOUND, result.getStatusCode());
     }
 
+    @Test
+    public void testAuthorizationRequestWithoutRedirectUri() {
+
+      Map<String, String> body = IntegrationTestUtils.getAuthorizationCodeTokenMap(serverRunning,
+          testAccounts,
+          "login",
+          "loginsecret",
```
Comment (strehle) [inline uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/AuthorizationCodeGrantIntegrationTests.java:203]: this client has only one redirect uri therefore you can omit it.

File: uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/AuthorizationCodeGrantIntegrationTests.java [hunk 3]
```diff
@@ -192,6 +194,34 @@ public void testZoneInactive() {
...
+          null,
+          null,
+          null,
+          false);
+
+      assertNotNull("Token not received", body.get("access_token"));
+
+      try {
+        IntegrationTestUtils.getAuthorizationCodeTokenMap(serverRunning, testAccounts, "app", "appclientsecret",
+            testAccounts.getUserName(), testAccounts.getPassword(),
```
Comment (strehle) [inline uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/AuthorizationCodeGrantIntegrationTests.java:216]: the app client has 2 entries for redirect-uri thus here it does not work.

These tests simply should help to know what we have already, nothing new

**Review by peterhaochen47 — APPROVED** [review]
